src/coreclr/vm/method.inl

```diff
@@ -19,6 +19,12 @@ inline bool MethodDesc::IsEligibleForTieredCompilation()
 inline bool MethodDesc::IsEligibleForTieredCompilation_NoCheckMethodDescChunk()
```

Comment:
This return value of this method is expected to be stable. `m_interpreterCode` is changing over the lifetime of MethodDesc, so this method won't return a stable result anymore.

I think this should be fixed in some other place.

Reply:
> Maybe we expose an API to ask the interpreter whether it plans to compile a method or not, which will query the config vars.

Config vars are just a testing tool. We can do simplest possible solution for that allows us to tests, like disable tiered compilation when any interpreter testing is on.

If we ever shipped interpreter with JIT, interpreter would be tier-0 and JIT would be higher tiers. It would be a bunch of work to make this system work and tune it well. I would not worry about interpreter + tiered compilation combo for now.